bin/spice/pkg/context/context.go

```diff
@@ -394,6 +394,19 @@ func (c *RuntimeContext) SetHttpEndpoint(endpoint string) {
 	c.httpEndpoint = endpoint
 }
 
+func (c *RuntimeContext) IsDownloadedFromHomebrew() bool {
+	executableDir, err := os.Executable()
+	if err != nil {
+		slog.Error("getting path", "error", err)
+		return false
+	}
+
+	spiceBinDir := filepath.Join(c.SpiceRuntimeDir(), "bin")
+	releaseFilePath := filepath.Join(spiceBinDir, constants.SpiceCliFilename)
+
+	return executableDir != releaseFilePath
```

Comment:
This should also check that `homebrew` is in the path - otherwise this would incorrectly report it was installed from homebrew when it could have just been installed into a non-standard path.

Reply:
If we detect a non-standard path that isn't homebrew, the message should be more generic, i.e. "Spice cannot upgrade because the executable is in a non-standard location: {location}."